compiler/driver.c

```diff
@@ -140,8 +140,19 @@ static int read_options(struct options * o, int argc, char * argv[]) {
                 continue;
             }
             if (eq(s, "-n") || eq(s, "-name")) {
+                /* Copy here so we do not free an argv item. */
+                const char * name = NULL;
+                char * new_name = NULL;
+                size_t len = 0;
+
                 check_lim(i, argc);
-                o->name = argv[i++];
+                name = argv[i++];
+
+                len = strlen(name);
+                new_name = malloc(len);
+                memcpy(new_name, name, len);
```

Comment:
These three lines can be replaced with `strdup` usage, can't they? Also `strdup` will make sure the new string ends with null byte, which your code doesn't seem to do.

Reply:
@mitya57 can `strdup` somehow be used with `-std=c90`?

Reply:
I'd probably avoid `strdup()` - it's specified by POSIX but not part of ISO C so it potentially limits portability for marginal convenience.